app/services/DataPointMapper.java

```diff
@@ -91,6 +93,14 @@ void mapDataPoint(final DataPointRecord dataPointRecord, EntityManager entityMan
         DimensionalDataSet dataSet = findOrCreateDataset(dataPointRecord.getDatasetID(), dataPointRecord.getS3URL(), entityManager);
 
         inputCSVParser.parseRowdataDirectToTables(entityManager, rowDataArray, dataSet);
+        createDatasetRowIndex(dataPointRecord.getDatasetID(), dataPointRecord.getIndex(),entityManager);
+    }
+
+    private void createDatasetRowIndex(UUID datasetId, long rowIndex, EntityManager entityManager) {
+        DimensionalDataSetRowIndex dataSetRowIndex = new DimensionalDataSetRowIndex();
+        dataSetRowIndex.setDatasetId(datasetId);
+        dataSetRowIndex.setRowIndex(rowIndex);
+        entityManager.persist(dataSetRowIndex);
```

Comment:
I guess these records won't be visible to the status checking transactions until the entire batch commits? If batches are large, that could be quite a long gap without any status updates.